### PR TITLE
Fix playwright tests for the discover section

### PR DESCRIPTION
**Pull request by elbywan**
#### This PR attempts to fix the following issues:

- A dummy static file server is spawned with a fixed 3001 port, which is an issue for machines running several jobs at the same time
- Disparities between snapshots taken on the github/ledger hosted runners (scroll bar visible/hidden inside the webview)

#### Fixes

- Always spawn the server on a free port, and inject the mocked manifest which contains the server port through an env variable.
- Use `*::-webkit-scrollbar {display: none}` to hide the scrollbar on all environments.

### ❓ Context

- **Impacted projects**: `LLD/LLC` 
- **Linked resource(s)**: `N/A` 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: libs/ledger-live-common/src/platform/providers/RemoteLiveAppProvider/api/index.ts [hunk 1]
```diff
@@ -34,6 +34,12 @@ const api = {
     provider: string
   ): Promise<LiveAppManifest[]> => {
     if (getEnv("MOCK")) {
+      if (getEnv("MOCK_REMOTE_LIVE_MANIFEST")) {
+        return [
+          ...mockData,
+          ...JSON.parse(getEnv("MOCK_REMOTE_LIVE_MANIFEST")),
```
Comment (ggilchrist-ledger) [inline libs/ledger-live-common/src/platform/providers/RemoteLiveAppProvider/api/index.ts:40]: OK how the hell does this work 😂

Comment (ggilchrist-ledger) [reply libs/ledger-live-common/src/platform/providers/RemoteLiveAppProvider/api/index.ts:40]: Ah, we pass something into the envvar? What exactly is it?

File: apps/ledger-live-desktop/tests/specs/services/discover.spec.ts [hunk 2]
```diff
@@ -15,16 +16,27 @@ let continueTest = false;
...
   try {
-    const response = await request.get("http://localhost:3001");
-    if (response.ok() === true) {
+    const port = await server.start();
+    const response = await request.get(`http://localhost:${port}`);
+    if (response.ok()) {
       continueTest = true;
-      console.info("========> Dummy test app successfully running on port 3001! <=========");
+      console.info(`========> Dummy test app successfully running on port ${port}! <=========`);
+      process.env.MOCK_REMOTE_LIVE_MANIFEST = JSON.stringify(server.manifest(port));
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-desktop/tests/specs/services/discover.spec.ts:24]: So this is where the magic happens by passing the manifest when starting the server?

File: apps/ledger-live-desktop/tests/utils/serve-dummy-app.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import handler from "serve-handler";
+import http from "http";
+import path from "path";
+
+export const server = http.createServer((request, response) => {
+  // You pass two more arguments for config and middleware
+  // More details here: https://github.com/vercel/serve-handler#options
+  return handler(request, response, {
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-desktop/tests/utils/serve-dummy-app.ts:8]: Not sure what is happening here by returning a handler or the `public` attribute. Does using `return` not kill the server?

File: apps/ledger-live-desktop/tests/utils/serve-dummy-app.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    public: path.resolve(__dirname, "dummy-app-build"),
+  });
+});
+
+export const start = (port = 0): Promise<number> => {
+  return new Promise((resolve, reject) => {
+    server
+      .listen(port, "localhost")
```
Comment (ggilchrist-ledger) [inline apps/ledger-live-desktop/tests/utils/serve-dummy-app.ts:16]: Does it just automatically find an empty port?

File: apps/ledger-live-desktop/tests/utils/serve-dummy-app.ts (hunk 3 above)
Comment (ggilchrist-ledger) [reply apps/ledger-live-desktop/tests/utils/serve-dummy-app.ts:8]: Ok makes more sense: https://www.npmjs.com/package/serve-handler

**Review by ggilchrist-ledger — COMMENTED** [review]
Looks good!

File: apps/ledger-live-desktop/tests/utils/dummy-app-build/index.html [hunk 5]
```diff
@@ -1 +1,24 @@
-<!doctype html><html lang="en"><head><meta charset="utf-8"/><link rel="icon" href="/favicon.ico"/><meta name="viewport" content="width=device-width,initial-scale=1"/><meta name="theme-color" content="#000000"/><meta name="description" content="Ledger Live App created using create-react-app"/><title>Ledger Live App</title><script defer="defer" src="/static/js/main.fbd28f85.js"></script><link href="/static/css/main.0419f816.css" rel="stylesheet"></head><body><noscript>You need to enable JavaScript to run this app.</noscript><div id="root"></div></body></html>
\ No newline at end of file
+<!doctype html>
+<html lang="en">
+
+<head>
+  <meta charset="utf-8" />
+  <link rel="icon" href="/favicon.ico" />
+  <meta name="viewport" content="width=device-width,initial-scale=1" />
```
[the remark this answers is not in the document]
Comment (elbywan) [reply apps/ledger-live-desktop/tests/utils/dummy-app-build/index.html:7]: no it's below: `*::-webkit-scrollbar {display: none}`

[the remark this answers is not in the document]
Comment (elbywan) [reply apps/ledger-live-desktop/tests/utils/dummy-app-build/index.html:7]: the rest was already there, but minified

**Pull request merged**